### ISSUE TITLE
Products appearing grayed-out in the shopfront, despite having stock set to zero

### ISSUE THREAD

**Issue by filipefurtad0** [openfoodfoundation/openfoodnetwork#8038]
## Description

**Bug report -slash- feature request**

A supplier sets the stock to zero of an item which was added to the OC by a hub. The item still appears in the shopfront, although grayed out and impossible to add to cart.

Despite the inconsistent behavior [in regards to the user guide](https://guide.openfoodnetwork.org/shopping-with-open-food-network/frequently-asked-questions#are-all-the-items-i-see-in-stock) this is a [potentially useful feature](https://openfoodnetwork.slack.com/archives/C02TZ6X00/p1628757766030500?thread_ts=1628757556.030000&cid=C02TZ6X00) although not explicit.

Also there might be some additional context needed to reproduce this unambiguously.

## Expected Behavior

Define this behavior in an explicit way:

a) show 0 stock items in the shopfront or
b) hide zero stock items from the shopfront

## Actual Behaviour

In a "simple" OC and enterprise relationship - ex. products supplied and OC coordinated from the same enterprise - a zero-stock item is not appearing in the shopfront.

If a supplier sets the stock to zero of an item which was added to the OC by a hub the item still appears in the shopfront, although grayed out and impossible to add to cart.

## Steps to Reproduce

Please note: An additional step might be missing to reproduce this. It could relate to products added to the inventory.

1. As a supplier: Create an item with stock and grant permissions to an OC coordinating hub to add products to their OC
2. As the coordinating hub: Add the products to the OC
3. As a customer: check the shopfront and see these items are available
4. As the supplier: set the stock to zero on one of these items
5. As the customer: notice the items are still appearing in the shopfront but appear grayed out

## Animated Gif/Screenshot

![Peek 2021-08-12 09-24](https://user-images.githubusercontent.com/49817236/129169293-8a6d6dbc-79f1-474d-ae76-cd7b49f3ca8a.gif)

## Workaround

None needed. Zero-stocked items cannot be added to the cart - which is the correct behavior.

## Severity

Adding the s4 label due to the inconsistency with the user guide.

## Your Environment

* Version used: v4.0.0
* Browser name and version: Firefox 90
* Operating System and version (desktop or mobile): Ubuntu 20.04

**Comment by drummer83** [comment]
If we ever work on this, here are some things to consider:
1. some users want sold out products (stock level zero) to be displayed in the shopfront to present their full product range. Those products should then be greyed out and could even have a label 'sold out'.
2. some users don't want products with stock level zero displayed in the shopfront at all because they use this as an easy way to select the products which should be visible in the order cycle or not. [See description here in Slack](https://openfoodnetwork.slack.com/archives/CG7NJ966B/p1636216193158100?thread_ts=1635896562.107500&cid=CG7NJ966B). 

One should consider making this an individual setting for each shopfront.

BUT: The 2nd use case is just a workaround for obviously inconvenient user experience when setting up an order cycle. So instead of making the greying out optional, one should improve the process of setting up the order cycles in the first place.

**Comment by BethanOFN** [comment]
User feedback from a large UK hub - it is confusing to show out of stock items in the shopfront when you have multiple products as the OOS items clog up the product list making it harder to navigate. I would vote for making this an optional shop setting, or at the least changing the order of the product list such that OOS items are shown last so that users don't have to scroll through loads of OOS items.

**Comment by RachL** [comment]
Yes handling sold out is something we are working on as specs here (future feature available for funding - WIP):

https://github.com/openfoodfoundation/wishlist/issues/147

In this issue 8038, we just want to fix the bug: the products should not be appearing (current correct behavior).

I'm editing my previous comment!
